### ISSUE TITLE
[Bug] borderColor hides bar color with large data

### ISSUE THREAD

**Issue by nico-robert** [apache/echarts#18185]
### Version

5.4.1

### Steps to Reproduce

I don't know if in version 5.2.1 It was the good result , but when I add `borderColor` property in ìtemStyle` (series), the color is not visible with large data, in 5.4.1 yes. So it hides the color of the bars.
```
    {
      name: 'Volumn',
      type: 'bar',
      xAxisIndex: 1,
      yAxisIndex: 1,
      itemStyle: {
        color: '#7fbe9e',
        // Add black color...
        borderColor: 'black'
      },
      large: true,
      encode: {
        x: 0,
        y: 5
      }
    }
```

### Current Behavior

In [5.4.1](https://echarts.apache.org/examples/en/editor.html?c=candlestick-large&code=MYewdgzgLgBArgBwMIgDYgE4wLwwOQDEApsAAzml4DcAUKJLIgEKYAmRGK6WuhAHAEEKlWvWgxWIAO5gumHPgLlWAQwDMwanXDjJMlhnac083ktJ8AYgCY-WsbFVQVKOGFi5rRAKyidjlWcFAHMiMA5AogB5AAkAGSQACicXEDcoAEpaEAQoAEtwBQBvGhgJQJUIIigALhgSsrKINIxgIjqU0pgAXwAaLvyoVHb6rrKoIgAPWvwAEQqYAQBbNPc6vBgAahgSAAsVDCgIADoAM0wlwOOVVlYUJcuIZIrXdwyuvoGQNHyEOobGlAMHlgqEMOsVJM8hA8P1GjBIdCAAogPLuDj_MaAgCeCBGeFQaKIeCx3Q-cJgUG-qAARiBJpj4aciIE4BgRgD4SkAFrfJaM-GNbECKEQACSYHYDJgpxUqCqWLKZPhyp6FOCwNYdQA2ljOY1hqcZngAIykACksMVMGBwV2xrNlop8LpUCp_Jg1nIpOdo0FMENDotVv9tvt60dIcFuyIIPDMD4pF9ZVd7rqidJXQAuhTJiLoTq9daoLj8cBIsFMNioy7VqoMNiAOIqP4yuVVZMwAD0XZgaIAbhwqnUgXAiJ3ERA4kT_jBwNyOCA6rL5UQ1daIAhCVBp-FZxBdtJl-2159_Us0esUgBZNE1xqXaV4G-Qkkq3368al9bliaVhv3mUGp5KwEpSnUJqdnSbj1k2LbHquE6iruHJzmAC4YEubaruu_qTgAKnkwAANb7oeUgIVUuGCpu24oWRR7YVRZ6CpOcQqDSRCoAxFFMaenYXmAV4VLeYCATAj7Cc416vpmZQ5l0wqioW8KfjAEDlsMI4YGOna0XkUACOyKgCv6B6MaORDWqqSofhumkjJZnbAaBkpTBBelbgZABycBLJx4Kekh0LsZx3H1Op5GUfx1psTOEXmbxK7McFECESRPHRdR8L6Tu8VFJFjHJaeckwApZQ8nyKmNGpJZ4usaIQCBxKdnmopge5MDakmMAmuVNHOIcEFJtaYRar13rvkWZlRZSOnjtabXQh10rdb0vX9fCdX4hAhJGOJqYgB6Zp6YNMwnaNkrDaQpX9f20JwHKMmtvqiXRb6VTAkQ4pudKkFYqweRLGETXgHUABsvoIHkJDfdVqnWmU_ZymOHmIzAoDcHUiByBg1mdmpSMoyMAC0_3-mUmOYB00iyCYeP-jZZXkl0n0wxA8N-oK20_iokrDNARHEeJBlEEsADKJZaVz_pU4FOP052lP06QNMyLjSswHShgcLj2MIAYRga-j2tGyratgIbuv0_jl2gOwpmCtKI0UzA2I6pBMAACzrWo63WFm1k-tNgpgCowPrAAamgfliZ2PP4DSBziUtP3gb1nZKctv1o_6osS1LHLo3L6wEAA7KcnEAJwtSbbDW1jieoCoJHiSx8LNxgoTabpdsgA7MtO3ULsU-7MDeEHKrZjQ3S0KcbjAPkhShOEGCRLECSJKA6QZFzwwBEEuDalmtBlPvMCTJHxMKJs4RSDA8wTIkXomp7PUmlkXTn4JcATAo4OkBgAAKgmuQU-AZqha0qEQK-qAxwKBklAXYxw16SiOokXeICTRejAV_SBdJL7tgUHfRYGA17YkSF7T-Z9IGqGxAAJT5qEBQ2DwHnCwIkc-eQFCkCoH2GAAAeDGqwoB8LyJsTYu99RJyqLA-BuAZEwOvtsRByDUGSCWBg4BnpAEkwmuAso7CYCcMgQAKx4Xw8xQivaWIkVI60BDYEQG1KYrMChEiqJQXzDRWi9GkGON4TB5QGFMLXNsRRciiAGJ6FiRx7ZjjNEOBg6J58chhDFKwSYCDAhqLrB4nJXi0GaKCWoahBpIHAHQFUDJWTcCeMwjBfJSDCk-KCdYMpZQ8inGMZUkA1TMk4GwLgNJYAan2NllUogNSJHRKZuffsMclhgAUHE-U2o1BuJAYkM05AtgwHqd49BQTvDkA6d2Xs2o8AjNhPgXpVQbkEmkN9KADzdhxmeQ8hZcCll4EDvCHsXVPY-xgH7T0fzGgpG1HkNxR9RrAH2IcE47DLhQDOBcQIhFgaJDwNiXF2ISbXmvCTW4AAdMAuxdg1AeDUCAMJ1qJEIXAsJuAf4TAyBkTsmxVnOJGTUrMxwqSWDyJMIgrBn4cutFy-kTj1n8sFcK0V4rOXcu6nKkAQqRVivacq6V7ZtR3KmZktVGrFXaslV82OAr1UKrFaQCV_pQhQHFiCMAzxnDrTyOtKVjKeV4lGUar1KqDV8vWlQ856kXVYhPh8Lo7IoBsmWSkNhC8l7LMdc64IrqUjrRSCtdaIzYHrQNYWjGkzZhAzGYPZGWAmqZuiV04xBa5QwAAHylr6Uo1A4yaIuoUGTWZOwcINsSE21Agj22yLlN2nKvbcAmgHVxKiala3LOwOjXNv1W0wBuq7MoAB-cNYpukGsGcMv1RajoIAOGuKQBldgTrXCAbpzdxDslAIYdGXIKjag3VKGAei-r6rLRWo1gjcDFrlJ-xoB6TRQbKHUMmUG6gAu8iAG0JA2Bzm6Qgdk900gQHWlUWAVItZrgQH0gyeRBxwd6rMrEcaE0RrrTG7oQA
)

### Expected Behavior

The good result for me in [5.2.1](https://echarts.apache.org/examples/en/editor.html?c=candlestick-large&version=5.2.1&code=MYewdgzgLgBArgBwMIgDYgE4wLwwOQDEApsAAzml4DcAUKJLIgEKYAmRGK6WuhAHAEEKlWvWgxWIAO5gumHPgLlWAQwDMwanXDjJMlhnac083ktJ8AYgCY-WsbFVQVKOGFi5rRAKyidjlWcFAHMiMA5AogB5AAkAGSQACicXEDcoAEpaEAQoAEtwBQBvGhgJQJUIIigALhgSsrKINIxgIjqU0pgAXwAaLvyoVHb6rrKoIgAPWvwAEQqYAQBbNPc6vBgAahgSAAsVDCgIADoAM0wlwOOVVlYUJcuIZIrXdwyuvoGQNHyEOobGlAMHlgqEMOsVJM8hA8P1GjBIdCAAogPLuDj_MaAgCeCBGeFQaKIeCx3Q-cJgUG-qAARiBJpj4aciIE4BgRgD4SkAFrfJaM-GNbECKEQACSYHYDJgpxUqCqWLKZPhyp6FOCwNYdQA2ljOY1hqcZngAIykACksMVMGBwV2xrNlop8LpUCp_Jg1nIpOdo0FMENDotVv9tvt60dIcFuyIIPDMD4pF9ZVd7rqidJXQAuhTJiLoTq9daoLj8cBIsFMNioy7VqoMNiAOIqP4yuVVZMwAD0XZgaIAbhwqnUgXAiJ3ERA4kT_jBwNyOCA6rL5UQ1daIAhCVBp-FZxBdtJl-2159_Us0esUgBZNE1xqXaV4G-Qkkq3368al9bliaVhv3mUGp5KwEpSnUJqdnSbj1k2LbHquE6iruHJzmAC4YEubaruu_qTgAKnkwAANb7oeUgIVUuGCpu24oWRR7YVRZ6CpOcQqDSRCoAxFFMaenYXmAV4VLeYCATAj7Cc416vpmZQ5l0wqioW8KfjAEDlsMI4YGOna0XkUACOyKgCv6B6MaORDWqqSofhumkjJZnbAaBkpTBBelbgZABycBLJx4Kekh0LsZx3H1Op5GUfx1psTOEXmbxK7McFECESRPHRdR8L6Tu8VFJFjHJaeckwApZQ8nyKmNGpJZ4usaIQCBxKdnmopge5MDakmMAmuVNHOIcEFJtaYRar13rvkWZlRZSOnjtabXQh10rdb0vX9fCdX4hAhJGOJqYgB6Zp6YNMwnaNkrDaQpX9f20JwHKMmtvqiXRb6VTAkQ4pudKkFYqweRLGETXgHUABsvoIHkJDfdVqnWmU_ZymOHmIzAoDcHUiByBg1mdmpSMoyMAC0_3-mUmOYB00iyCYeP-jZZXkl0n0wxA8N-oK20_iokrDNARHEeJBlEEsADKJZaVz_pU4FOP052lP06QNMyLjSswHShgcLj2MIAYRga-j2tGyratgIbuv0_jl2gOwpmCtKI0UzA2I6pBMAACzrWo63WFm1k-tNgpgCowPrAAamgfliZ2PP4DSBziUtP3gb1nZKctv1o_6osS1LHLo3L6wEAA7KcnEAJwtSbbDW1jieoCoJHiSx8LNxgoTabpdsgA7MtO3ULsU-7MDeEHKrZjQ3S0KcbjAPkhShOEGCRLECSJKA6QZFzwwBEEuDalmtBlPvMCTJHxMKJs4RSDA8wTIkXomp7PUmlkXTn4JcATAo4OkBgAAKgmuQU-AZqha0qEQK-qAxwKBklAXYxw16SiOokXeICTRejAV_SBdJL7tgUHfRYGA17YkSF7T-Z9IGqGxAAJT5qEBQ2DwHnCwIkc-eQFCkCoH2GAAAeDGqwoB8LyJsTYu99RJyqLA-BuAZEwOvtsRByDUGSCWBg4BnpAEkwmuAso7CYCcMgQAKx4Xw8xQivaWIkVI60BDYEQG1KYrMChEiqJQXzDRWi9GkGON4TB5QGFMLXNsRRciiAGJ6FiRx7ZjjNEOBg6J58chhDFKwSYCDAhqLrB4nJXi0GaKCWoahBpIHAHQFUDJWTcCeMwjBfJSDCk-KCdYMpZQ8inGMZUkA1TMk4GwLgNJYAan2NllUogNSJHRKZuffsMclhgAUHE-U2o1BuJAYkM05AtgwHqd49BQTvDkA6d2Xs2o8AjNhPgXpVQbkEmkN9KADzdhxmeQ8hZcCll4EDvCHsXVPY-xgH7T0fzGgpG1HkNxR9RrAH2IcE47DLhQDOBcQIhFgaJDwNiXF2ISbXmvCTW4AAdMAuxdg1AeDUCAMJ1qJEIXAsJuAf4TAyBkTsmxVnOJGTUrMxwqSWDyJMIgrBn4cutFy-kTj1n8sFcK0V4rOXcu6nKkAQqRVivacq6V7ZtR3KmZktVGrFXaslV82OAr1UKrFaQCV_pQhQHFiCMAzxnDrTyOtKVjKeV4lGUar1KqDV8vWlQ856kXVYhPh8Lo7IoBsmWSkNhC8l7LMdc64IrqUjrRSCtdaIzYHrQNYWjGkzZhAzGYPZGWAmqZuiV04xBa5QwAAHylr6Uo1A4yaIuoUGTWZOwcINsSE21Agj22yLlN2nKvbcAmgHVxKiala3LOwOjXNv1W0wBuq7MoAB-cNYpukGsGcMv1RajoIAOGuKQBldgTrXCAbpzdxDslAIYdGXIKjag3VKGAei-r6rLRWo1gjcDFrlJ-xoB6TRQbKHUMmUG6gAu8iAG0JA2Bzm6Qgdk900gQHWlUWAVItZrgQH0gyeRBxwd6rMrEcaE0RrrTG7oQA)

### Environment

```markdown
- OS: macOS Ventura
- Browser: Safari 16.1
- Framework:
```

**Comment by Ovilia** [comment]
The current behavior seems to be reasonable. As for the candlestick series, we are considering provide a way to dynamically change the range of data one bar represents so that there won't be too many bars like the example. But for general series, I think the current behavior is as expected.

**Comment by nico-robert** [comment]
okay @Ovilia, It was just to show the difference between 2 versions of Echarts. For me (personal opinion) the result in version 5.2.1 is correct , because the borderColor does not interact with bar color with large data (I'm surprised). But I am agree with you for general series. Not a bug probably , but the visualization was more fun with this version. That said thank you for your quick response.

**Comment by Ovilia** [comment]
I seemed to misunderstand the demo to be a problem with candlestick series. This seems to be wrong with the bar series.

**Comment by kai890707** [comment]
@nico-robert 
If you need to solve the problem urgently, maybe it would be better to set borderWidth less than 1?

I will try to fix this bug.

**Comment by nico-robert** [comment]
@kai890707 , thanks 

> If you need to solve the problem urgently, maybe it would be better to set borderWidth less than 1?

Until something better it was my solution too.

I'm newbie... The difficulty label is set to `easy` (To tell you the truth, I tried again to find where the bug was today...) But I don’t know how to debug this lib and my knowledge of JavaScript/TypeScript is basic.
Please if your time permits, can you show me where this property is found in the source file ?

**Comment by Traumflug** [comment]
@[Ovilia](https://github.com/Ovilia)
> As for the candlestick series, we are considering provide a way to dynamically change the range of data one bar represents so that there won't be too many bars like the example.

I don't think that's worth the work, people working with trading charts wouldn't use it. Candlestick charts are less about displaying a lot of numbers nicely, they're more about learning how a market behaves. Accordingly, a chart showing hourly candles is fundamentally distinct from a chart showing daily candles. Switching between both while zooming would mess up the view.

Regarding improving candlesticks: a distinct color for just the wicks would be a nice addition.

**Comment by DevitX** [comment]
> I seemed to misunderstand the demo to be a problem with candlestick series. This seems to be wrong with the bar series.

I looked into this problem, as you mentioned the issue is in the bar series (BarView.ts). In version 5.3.0-rc.1, commit 897081e [fix(bar): use rect instead of line to draw large paths], the createLarge function sets el.style.stroke to a color with  el.useStyle(data.getVisual('style'));,. This causes the problem. If you reset el.style.stroke to null (default), the issue seems to be resolved and follows the behavior in 5.2.1. I’ve opened a pull request about this: #20465.
